File: scala-3/euler/src/main/scala/com/rbleggi/euler/P15.scala
package com.rbleggi.euler

@main def runP15(): Unit = {
  def factorial(n: BigInt): BigInt = if (n == 0) 1 else n * factorial(n - 1)

  val gridSize = 20
  val result = factorial(2 * gridSize) / (factorial(gridSize) * factorial(gridSize))

  println(s"The number of routes through a 20x20 grid is: $result")
}

File: kotlin/euler/src/main/kotlin/com/rbleggi/euler/P15.kt
package com.rbleggi.euler

import java.math.BigInteger

fun mainP15() {
    fun factorial(n: BigInteger): BigInteger =
        if (n == BigInteger.ZERO) BigInteger.ONE else n * factorial(n - BigInteger.ONE)

    val gridSize = BigInteger.valueOf(20)
    val result = factorial(gridSize * BigInteger.TWO) / (factorial(gridSize) * factorial(gridSize))

    println("The number of routes through a 20x20 grid is: $result")
}
